docs/vesting.md

```diff
@@ -0,0 +1,101 @@
+2. add_vesting_milestone(vesting_timestamp: u64, grantee: ContractAddress, amount: u128): This function adds a vesting milestone for a grantee at a specific vesting timestamp.
+3. add_linear_vesting_schedule(first_vest: u64, period: u64, increments_count: u64, total_amount: u128): This function adds a linear vesting schedule with specified parameters.
+
+
+
+###  Usage
+
+# Executing Vesting
+To execute the vesting process for a grantee at a specified timestamp, use the vest function. This function will check if the conditions are met and, if so, mint the vested tokens to the grantee.
+
+# Example usage of vesting tokens
+
+contract_instance.vest(
+    grantee=ContractAddress,
+    vested_timestamp=1650000000
+);
+
+# Parameters
+* grantee: The address of the grantee.
+* vested_timestamp: The timestamp of the vesting event.
+
+
+## Adding a Vesting Milestone
+
+To add a single vesting milestone, we use the add_vesting_milestone function. This function sets a specific amount of tokens to be vested at a given timestamp for a grantee.
+
+# Example usage of adding a vesting milestone
+
+contract_instance.add_vesting_milestone(
+    vesting_timestamp=1632555600,
+    grantee=ContractAddress,
+    amount=1000
+);
+
+# Parameters
+* vesting_timestamp: The timestamp when the vesting event should occur.
+* grantee: The address of the grantee.
+* amount: The amount of tokens to be vested.
+
+
+
+## Adding a Linear Vesting Schedule
+
+To add a linear vesting schedule, we use the add_linear_vesting_schedule function. This function divides the total amount to be vested into increments and schedules them over specified periods.
+
+# Example usage of adding a Linear vesting schedule
+
+contract_instance.add_linear_vesting_schedule(
+    first_vest=1650000000,
+    period=2592000,  // 30 days in seconds
+    increments_count=10,
+    total_amount=1200,
+    grantee=ContractAddress
+);
+
+# Parameters
+
+* first_vest: The timestamp of the first vesting event.
+* period: The time interval between vesting events.
+* increments_count: The number of vesting increments.
+* total_amount: The total amount of tokens to be vested.
+* grantee: The address of the grantee.
+
+
+
+## Customization
+
+To customize the Vesting contract for specific protocol’s need, use the methods defined below; 
+1. Custom Proposals: To add new vesting schedules, the component should be part of a custom proposal that calls either add_linear_vesting_schedule or add_vesting_milestone.
+2. Contract Integration: The vest function should be exported externally from the contract to allow protocols to execute the vesting process.
+
+
+
+## Components
+
+The Vesting contract is composed of the following components:
+
+1. VestingImpl: Implementation of the IVesting trait for managing vesting operations.
+2. Storage: Storage component for storing vesting milestones.
```

Comment:
What?? The whole vesting.cairo is technically a component. What is the point of this heading?